### PR TITLE
[nettle] Switch sources on windows away from SMP fork. 

### PR DESCRIPTION
**Pull request by Neumann-A**
includes #23466

### CODE REVIEW

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/23525.

**Comment by JackBoosY** [comment]
Close and reopen this PR to re-trigger the ci test.

**Review by JackBoosY — APPROVED** [review]

File: ports/nettle/portfile.cmake [hunk 1]
```diff
@@ -1,137 +1,96 @@
...
-    vcpkg_fixup_pkgconfig()
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share/")
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+vcpkg_fixup_pkgconfig()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share/")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-    # # Handle copyright
-    file(INSTALL "${SOURCE_PATH}/COPYINGv3" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+# # Handle copyright
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:91]:
```suggestion
# Handle copyright
```

File: ports/nettle/portfile.cmake [hunk 2]
```diff
@@ -1,137 +1,96 @@
...
-        OUT_SOURCE_PATH SOURCE_PATH
-        REPO nettle/nettle
-        REF  9e2bea82b9fb606bffd2d3f648e05248e146e54f #v3.6
-        SHA512 008089eba2ef197a0ec6a266baac485e72051e646d19861f3fb605915a591bc2dd38edcb4ea7eaad958ea5d56f7744d42c25b691b49921a1285edd22f9c90b7f
-        HEAD_REF master
-        PATCHES 
-		fix-InstallLibPath.patch
-		flags.patch
-    )
+set(tool_names des ecc ) # aes gcm sha twofish ?
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:55]:
```suggestion
set(tool_names des ecc ) # aes gcm sha twofish?

```

File: ports/nettle/portfile.cmake [hunk 3]
```diff
@@ -1,137 +1,96 @@
...
-        set(CONFIGURATION_DEBUG DebugDLL)
-    else()
-        set(CONFIGURATION_RELEASE Release)
-        set(CONFIGURATION_DEBUG Debug)
-    endif()
+vcpkg_from_gitlab(
+    GITLAB_URL https://git.lysator.liu.se/
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO nettle/nettle
+    REF  52bacacaf4339fd78289f58919732f1f35bea1c1 #v3.7.3
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:12]:
```suggestion
    REF 52bacacaf4339fd78289f58919732f1f35bea1c1 #v3.7.3

```

File: ports/nettle/portfile.cmake [hunk 4]
```diff
@@ -1,137 +1,96 @@
...
+    SHA512 0130d14195274eeec11e8299793e3037f4b84d8fb4b5c5c9392b63ee693ed5713434070744b1a44e14a6a5090d655917c1dd296e2011cd99e3c316ef5d8ee395
+    HEAD_REF master
+    PATCHES 
+    fix-InstallLibPath.patch
+    flags.patch
+    ${patches}
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:18]:
```suggestion
        fix-InstallLibPath.patch
        flags.patch
        ${patches}
```

File: ports/nettle/portfile.cmake [hunk 5]
```diff
@@ -1,137 +1,96 @@
...
-		fix-InstallLibPath.patch
-		flags.patch
-    )
+set(tool_names des ecc ) # aes gcm sha twofish ?
+list(TRANSFORM tool_names APPEND "data")
+list(TRANSFORM tool_names APPEND "${VCPKG_HOST_EXECUTABLE_SUFFIX}")
 
+if(VCPKG_CROSSCOMPILING)
+    list(TRANSFORM tool_names PREPEND "${CURRENT_HOST_INSTALLED_DIR}/manual-tools/${PORT}/")
+    file(COPY ${tool_names} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/" )
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:61]:
```suggestion
    file(COPY ${tool_names} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/")

```

File: ports/nettle/portfile.cmake [hunk 6]
```diff
@@ -1,137 +1,96 @@
...
+    if(NOT VCPKG_BUILD_TYPE)
+        file(COPY ${tool_names} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/" )
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:63]:
```suggestion
        file(COPY ${tool_names} DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/")

```

File: ports/nettle/portfile.cmake [hunk 7]
```diff
@@ -1,137 +1,96 @@
...
+endif()
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-        set(OPTIONS --disable-static)
-    else()
-        set(OPTIONS --disable-shared)
+        # def files are created by running 'llvm-nm <libname> | findstr /R /C:"[RT] _*nettle_"' on the static build and replacing '00[0-9abcdef]+ [RT]' with spaces
+        # please update the defs if the version is bumped
+        set(build_dir "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/")
+        configure_file("${CURRENT_PORT_DIR}/nettle-${VCPKG_TARGET_ARCHITECTURE}.def"  "${build_dir}/nettle.def" COPYONLY)
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:71]:
```suggestion
        configure_file("${CURRENT_PORT_DIR}/nettle-${VCPKG_TARGET_ARCHITECTURE}.def" "${build_dir}/nettle.def" COPYONLY)

```

File: ports/nettle/portfile.cmake [hunk 8]
```diff
@@ -1,137 +1,96 @@
...
+        configure_file("${CURRENT_PORT_DIR}/hogweed-${VCPKG_TARGET_ARCHITECTURE}.def"  "${build_dir}/hogweed.def" COPYONLY)
+        if(NOT VCPKG_BUILD_TYPE)
+            set(build_dir "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/")
+            configure_file("${CURRENT_PORT_DIR}/nettle-${VCPKG_TARGET_ARCHITECTURE}.def"  "${build_dir}/nettle.def" COPYONLY)
+            configure_file("${CURRENT_PORT_DIR}/hogweed-${VCPKG_TARGET_ARCHITECTURE}.def"  "${build_dir}/hogweed.def" COPYONLY)
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:76]:
```suggestion
            configure_file("${CURRENT_PORT_DIR}/nettle-${VCPKG_TARGET_ARCHITECTURE}.def" "${build_dir}/nettle.def" COPYONLY)
            configure_file("${CURRENT_PORT_DIR}/hogweed-${VCPKG_TARGET_ARCHITECTURE}.def" "${build_dir}/hogweed.def" COPYONLY)

```

File: ports/nettle/portfile.cmake [hunk 9]
```diff
@@ -1,137 +1,96 @@
...
-        SOURCE_PATH ${SOURCE_PATH}
-        AUTOCONFIG
-        OPTIONS
-            --disable-documentation
-            --disable-openssl
-            ${OPTIONS}
-    )
+if(NOT VCPKG_CROSSCOMPILING)
+    list(TRANSFORM tool_names PREPEND "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/")
+    file(COPY ${tool_names} DESTINATION "${CURRENT_PACKAGES_DIR}/manual-tools/${PORT}" )
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:84]:
```suggestion
    file(COPY ${tool_names} DESTINATION "${CURRENT_PACKAGES_DIR}/manual-tools/${PORT}")

```

File: ports/nettle/portfile.cmake [hunk 10]
```diff
@@ -1,137 +1,96 @@
...
-            name.dir.patch
-            runtime.patch
-            remove_gmpd.patch
-    )
-
-    include(${CURRENT_INSTALLED_DIR}/share/yasm-tool-helper/yasm-tool-helper.cmake)
-    yasm_tool_helper(OUT_VAR YASM)
-    file(TO_NATIVE_PATH "${YASM}" YASM)
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+    set(patches yasm.patch      # the asm changes are a downgrade to an older version
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:2]:
```suggestion
    set(patches yasm.patch # the asm changes are a downgrade to an older version

```

File: ports/nettle/portfile.cmake [hunk 11]
```diff
@@ -1,137 +1,96 @@
...
+    fix-InstallLibPath.patch
+    flags.patch
+    ${patches}
+)
 
-    if(VCPKG_TARGET_IS_UWP)
-        string(APPEND CONFIGURATION_RELEASE WinRT)
-        string(APPEND CONFIGURATION_DEBUG WinRT)
-    endif()
+if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (Thomas1664) [inline ports/nettle/portfile.cmake:21]:
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)

```

**Review by Thomas1664 — COMMENTED** [review]
Some very small format **suggestions**.

File: ports/nettle/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/nettle/portfile.cmake:91]: Just remove that line...

File: ports/nettle/portfile.cmake (hunk 11 above)
Comment (dg0yt) [reply ports/nettle/portfile.cmake:21]: `"dynamic`"

**Comment by Thomas1664** [comment]
> Some very small format **suggestions**.

They don't _have_ to be fixed before merge

**Comment by Neumann-A** [comment]
> They don't have to be fixed before merge

I probably would just commit them if i wouldn't have to re-pull and run `vcpkg x-add-version` manually again.  (So that issue should probably be fixed first ;) )

**Comment by strega-nil-ms** [comment]
Thanks y'all!

**Pull request merged**